src/main/java/com/kryptnostic/rhizome/mapstores/cassandra/DefaultCassandraMapStore.java

```diff
@@ -109,19 +111,25 @@ public void delete( K key ) {
 
     @Override
     public void deleteAll( Collection<K> keys ) {
-        session.execute( DELETE_ALL_QUERY.bind( keys ) );
+        List<String> mappedKeys = new ArrayList<>( keys.size() );
+        for ( K key : keys ) {
+            mappedKeys.add( keyMapper.fromKey( key ) );
+        }
+        session.execute( DELETE_ALL_QUERY.bind( mappedKeys ) );
     }
 
     @Override
     public K generateTestKey() {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException( "THIS METHOD HAS NOT BEEN IMPLEMENTED, BLAME Drew Bailey drew@kryptnostic.com" );
+        throw new UnsupportedOperationException(
```

Comment:
Yes, though they'll still just throw an exception since they rely on being able to generate a specific datatype. They really should not actually be part of the Mapstores at all but we haven't had time to pull them out yet.